xbmc/pvr/addons/PVRClients.cpp

```diff
@@ -417,6 +417,35 @@ int CPVRClients::EnabledClientAmount() const
   return iReturn;
 }
 
+std::vector<SPVRClientInfo> CPVRClients::GetEnabledClientInfos() const
+{
+  std::vector<SPVRClientInfo> clientInfos;
+
+  CPVRClientMap clientMap;
+  {
+    CSingleLock lock(m_critSection);
+    clientMap = m_clientMap;
```

Comment:
I'm not sure whether `m_clientMap` contains only enabled items. Could you please double-check. Maybe we can avoid calling addonmgr here.

Reply:
You're right. It's only enabled add-ons. I used the logic from the function above this one which must also be incorrect. So I guess that can be fixed too.

https://github.com/xbmc/xbmc/blob/5b2911371eccd8b9bacaa9eb61cd08575a59f84c/xbmc/pvr/addons/PVRClients.cpp#L401-L418

Reply:
> You're right. It's only enabled add-ons. ... So I guess that can be fixed too.

As I said, I'm not 100% sure... we should be careful when changing things here. This is mission critical code. ;-)

Reply:
Seems you are right, m_clientMap does only contain enabled addons.

Reply:
True, I will only change the function I added.

Reply:
Fixing the other function is not related to this PR anyway.

Reply:
I switched this to GetAddonInfo which is much lighter weight. It is still needed as I need to get the name of the addon.